raiden/network/rpc/client.py

```diff
@@ -474,7 +495,14 @@ def approve(self, contract_address, allowance):
             startgas=self.startgas,
             gasprice=self.gasprice,
         )
-        self.client.poll(transaction_hash.decode('hex'), timeout=self.poll_timeout)
+
+        try:
+            self.client.poll(transaction_hash.decode('hex'), timeout=self.poll_timeout)
+        except JSONRPCPollTimeoutException:
+            raise BlockchainPollTimeout()
+        except InvalidTransaction:
+            raise BlockchainInvalidTransaction()
```

Comment:
I added the Raiden-namespaced exception for easier catching of errors,
and that you don't have to import different exceptions from different modules (`ethereum.exceptions`and `pyethapp.rpc_client`).
We could modify the "RaidenException"-child to include the information and traceback of the original Exception.
It would be simpler to just re-raise, and it is most likely only my feeling that this is a little ugly.
But do you agree that the checking of the exceptions in the code serves documentation purposes?